File: app/src/main/java/com/motorola/cdeives/motofretado/http/jsonapi/ErrorDocument.java
package com.motorola.cdeives.motofretado.http.jsonapi;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.motorola.cdeives.motofretado.http.Error;

public class ErrorDocument {
    @Nullable ErrorData[] errors;

    public @NonNull Error toError() throws IllegalStateException {
        if (errors == null || errors.length == 0) {
            throw new IllegalStateException("no error found in document");
        }

        Error error = new Error();

        error.status = errors[0].status;
        error.title = errors[0].title;
        error.details = errors[0].detail;

        return error;
    }
}

File: app/src/main/java/com/gmail/cristiandeives/motofretado/http/jsonapi/ErrorDocument.kt
package com.gmail.cristiandeives.motofretado.http.jsonapi

import com.gmail.cristiandeives.motofretado.http.Error

internal data class ErrorDocument(val jsonapi: JSONAPI?, val errors: List<ErrorData>, val links: Links?) {
    internal fun toError(): Error {
        jsonapi?.validateVersion()
        val firstError = errors.firstOrNull() ?: throw IllegalStateException("no error found in document")

        return Error(firstError.status, firstError.title, firstError.detail)
    }
}